tests/test_read_boundless.py

```diff
@@ -11,6 +13,7 @@ def test_read_boundless_natural_extent():
     with rasterio.open('tests/data/RGB.byte.tif') as src:
         data = src.read(boundless=True)
         assert data.shape == (3, src.height, src.width)
+        assert data[0].mean() == src.read(1).mean()
```

Comment:
@sgillies this seems potentially risky across platforms since it is a floating point precision equality comparison.  What about

```assert abs(data[0].mean() - src.read(1).mean()) < 0.0001```